client/ayon_speedtree/api/pipeline.py

```diff
@@ -0,0 +1,326 @@
+"""Pipeline tools for AYON Speedtree integration."""
+import os
+import ast
+import json
+import shutil
+import logging
+import tempfile
+import pyblish.api
+from ayon_core.host import HostBase, IWorkfileHost, ILoadHost, IPublishHost
+from ayon_core.pipeline import (
+    register_creator_plugin_path,
+    register_loader_plugin_path,
+    AYON_CONTAINER_ID,
+    registered_host
+)
+from ayon_core.pipeline.context_tools import get_global_context
+
+from ayon_core.settings import get_current_project_settings
+from ayon_core.lib import register_event_callback
+from ayon_core.tools.utils import host_tools
+from ayon_speedtree import SPTREE_ADDON_ROOT
+from .lib import get_workdir, execute_sptree_command
+
+import SpeedTree
+
+
+METADATA_SECTION = "avalon"
+SPTREE_SECTION_NAME_CONTEXT = "context"
+SPTREE_METADATA_CREATE_CONTEXT = "create_context"
+SPTREE_SECTION_NAME_INSTANCES = "instances"
+SPTREE_SECTION_NAME_CONTAINERS = "containers"
+
+
+log = logging.getLogger("ayon.hosts.speedtree")
+
+
+class SpeedtreeHost(HostBase, IWorkfileHost, ILoadHost, IPublishHost):
+    name = "speedtree"
+
+    @staticmethod
+    def show_tools_dialog():
+        """Show tools dialog with actions leading to show other tools."""
+        show_tools_dialog()
+
+    def install(self):
+        # Create workdir folder if does not exist yet
+        workdir = os.getenv("AYON_WORKDIR")
+        if not os.path.exists(workdir):
+            os.makedirs(workdir)
```

Comment:
How come this is up to the `install()` logic? Seems weird, the workdir should really be created already before the host has been started. This seems wrong?

Reply:
it shouldn't be there, I should do more proper clean-up